### ISSUE TITLE
Resolve VRT issues for local runs

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#5824]
## Bug Description

We recently updated our VRT infrastructure to support running on ARM/M1 machines which are becoming more common rather quickly.

While the build runs on these machines where it didn't before, it seems to have introduced some degree of instability in local runs on Intel machines (and M1s?).

At current, I see ~10 scenario failures when running locally, most of which seem to be very small differences in font rendering although some appear to have timeouts as well, albeit inconsistently.

I do see some amount of failures consistently though. See https://github.com/google/site-kit-wp/issues/4619#issuecomment-1241116909 

## Steps to reproduce

1. Run `npm run test:visualtest` locally
2. See failures

## Additional Context

 - OS: MacOS 12.5.1

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Visual regression tests should consistently run successfully, both locally (on Intel + ARM) and on GitHub actions

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by asvinb** [comment]
Update after spending some time on the ticket:

- Have yet to determine the cause of the issue
- Tried loading fonts within Storybook differently, using webfontloader and trigger backstop when the fonts are loaded via the custom `wf-active` class name.
- Might not be related to whether fonts are already loaded.

**Comment by eclarke1** [comment]
@FlicHollis I think we can remove this from Sprint 85 as it looks like it needs more work on the IB :-)

**Comment by FlicHollis** [comment]
This has been removed, thanks @eclarke1

**Comment by asvinb** [comment]
Latest updates:

- Fonts are not rendering even though technically they are loaded.
Reference:
![image](https://user-images.githubusercontent.com/31818/192775444-a5e5546a-a5c4-4d79-bcf6-ed08cb3ee9fe.png)

Test:
![image](https://user-images.githubusercontent.com/31818/192775503-8536cae5-8266-4a06-a0e2-9ba11ae5eb59.png)
- No improvements noted when increasing the `delay` in `tests/backstop/config.js`
- The same tests consistently fail (for e.g Global admin bar) which means that potentially we are missing something in those tests.

**Comment by techanvil** [comment]
I am also seeing failures on my Linux workstation(Ubuntu 22.04), similar to the description but more, ~30 failures here.

**Comment by asvinb** [comment]
@aaemnnosttv Not sure if am missing something here and how the GH actions are all passing, but if you look at one of the [reference image on develop](https://github.com/google/site-kit-wp/blob/develop/tests/backstop/reference/google-site-kit_Settings_Connected_Services_0_document_0_small.png) and you compare it with what [we currently have in Storybook for that image](https://google.github.io/site-kit-wp/storybook/develop/?path=/story/settings--connected-services), they are different, with the GA4 CTA present in Storybook and not in the reference image.
Screenshots in case things change:

Storybook:
<img width="1001" alt="image" src="https://user-images.githubusercontent.com/31818/199775678-79c2378b-df7a-4f79-8cdc-814807034c91.png">

BackstopJS Reference:
<img width="1080" alt="image" src="https://user-images.githubusercontent.com/31818/199775782-df5f65c6-50df-4fe5-bf22-ba6650ee29f1.png">

Weird how it VRT fails locally on `develop` whereas in GH actions, everything works fine 🤔 
Do you have any idea @aaemnnosttv @tofumatt @eugene-manuilov ?

**Comment by aaemnnosttv** [comment]
Wow, that is indeed strange. Perhaps @techanvil or @nfmohit could investigate this further?

**Comment by techanvil** [comment]
Having discovered that the VRT suite is not actually running as expected in CI, I am parking this issue until that has been resolved: https://github.com/google/site-kit-wp/issues/6151

**Comment by techanvil** [comment]
Based on the work on #6151, it looks like this issue will still be relevant, with a reduction in the number of test failures but still a couple of cases that need fixing and also a more general issue of stability with unpredictable timeouts occurring locally.

**Comment by derweili** [comment]
@asvinb @aaemnnosttv 
We fixed the VRT font loading issue with #6324 
So I think we can close this issue.

**Comment by techanvil** [comment]
I think it could be worth keeping to see about fixing the timeouts that occur when running locally...

**Comment by aaemnnosttv** [comment]
I tried running on develop today and most of them failed for me with what seemed to be impercievable font-related differences.

```
       report | Test completed...
       report | 16 Passed
       report | 152 Failed
```

E.g.

<img width="967" alt="image" src="https://user-images.githubusercontent.com/1621608/212743536-72a09814-9b47-43b6-8d7a-ec12e6876b3f.png">

When scrubbing the diff, I can't really notice a difference 🤔

**Comment by derweili** [comment]
@aaemnnosttv what machine do you use? For me on mac M1 the VRT tests do not fail.

**Comment by asvinb** [comment]
@aaemnnosttv @derweili I just checked locally (Mac with Intel chip) and there are no failures:
<img width="718" alt="image" src="https://user-images.githubusercontent.com/31818/212847042-42489894-ad6c-403a-9c72-87c673280528.png">

I see images have been updated last night, can you try again @aaemnnosttv ?

**Comment by derweili** [comment]
@asvinb the updates were only about removing unused reference images. This should not change anything for this issue.

**Comment by aaemnnosttv** [comment]
That's odd @asvinb – I'm also on a Intel based Mac. I ran it again on develop just now and got the same results 😞

**Comment by derweili** [comment]
@aaemnnosttv is you VRT docker image up to date? Maybe you could try rebuilding it.

**Comment by aaemnnosttv** [comment]
@derweili I deleted the images I had and ran it again and everything's passing now 🤷‍♂️ 🎉 

I'll close this out then but do you know what changed in the image? It's built based on the specific version of backstop but something must have changed 🤔

**Comment by derweili** [comment]
@aaemnnosttv actually I don't really know. With #6324 only images were updated that had visual changes, mostly due to font rendering errors.

The VRT image from your example was updated 1 month ago but also without visual changes. https://github.com/google/site-kit-wp/pull/6303/files

However, I experienced the error you had (errors without visual change), when I ran backstopjs locally without docker. I did this to debug font loading. So backstop used a local instance of chromium. When running the VRT tests this way, almost all the tests failed even though there were no visual changes.
So I thought, this could be related to a different font rendering/antialising setting or so.

So I though it might be related to an outdated VRT docker image with different settings.

But if deleting the test images helped for you, should be delete the test images before the test? Why do we need to keep the test images for all test runs?
